deepchem/data/dft_data.py

```diff
@@ -0,0 +1,319 @@
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        assert len(self.get_systems()) == 1, "dens entry can only have 1 system"
+        self._grid: Optional[BaseGrid] = None
+
+    @property
+    def entry_type(self) -> str:
+        return "dens"
+
+    def _get_true_val(self) -> torch.Tensor:
+        # get the density profile from PySCF's CCSD calculation
+
+        system = self.get_systems()[0]
+        dens = np.load(self["trueval"])
+        true_val = torch.from_numpy(dens)
+        return torch.as_tensor(true_val, device=self.device)
+
+    def get_val(self, qcs: List[KSCalc]) -> torch.Tensor:
+        qc = qcs[0]
+
+        # get the integration grid infos
+        grid = self._get_integration_grid()
+        rgrid = grid.get_rgrid()
+
+        # get the density profile
+        return qc.dens(rgrid)
+
+    def get_integration_grid(self) -> BaseGrid:
+        if self._grid is None:
+            system = self.get_systems()[0]
+
+            # reduce the level of grid
+            orig_grid_level: Optional[int] = None
+            if "grid" in system:
+                orig_grid_level = system["grid"]
+
+            # get the dqc grid
+            system["grid"] = "sg2"
+            dqc_mol = system.get_dqc_system()
+            dqc_mol.setup_grid()
+            grid = dqc_mol.get_grid()
+            assert grid.coord_type == "cart"
+
+            # restore the grid level
+            if orig_grid_level is not None:
+                system["grid"] = orig_grid_level
+            else:
+                system.pop("grid")
+
+            self._grid = grid
+
+        return self._grid
+
+
+class EntryForce(DFTEntry):
+    """Entry for force at the experimental equilibrium position"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        assert len(
+            self.get_systems()) == 1, "force entry can only have 1 system"
+
+    @property
+    def entry_type(self) -> str:
+        return "force"
+
+    def _get_true_val(self) -> torch.Tensor:
+        # get the density matrix from PySCF's CCSD calculation
+        return torch.tensor(0.0, dtype=self.dtype, device=self.device)
+
+    def get_val(self, qcs: List[KSCalc]) -> torch.Tensor:
+        return qcs[0].force()
+
+
+class EntryIE(DFTEntry):
+    """Entry for Ionization Energy (IE)"""
+
+    @property
+    def entry_type(self) -> str:
+        return "ie"
+
+    def _get_true_val(self) -> torch.Tensor:
+        return torch.as_tensor(self["true_val"],
+                               dtype=self.dtype,
+                               device=self.device)
+
+    def get_val(self, qcs: List[KSCalc]) -> torch.Tensor:
+        glob = {"systems": qcs, "energy": self.energy}
+        return eval(self["cmd"], glob)
+
+
+class EntryAE(EntryIE):
+    """Entry for Atomization Energy (AE)"""
+
+    @property
+    def entry_type(self) -> str:
+        return "ae"
+
+
+def load_entries(entry_path, device):
```

Comment:
This should be a `DataLoader` class. See `CSVLoader`, `JSONLoader`, etc. We are effectively defining a new filetype that we are adding to DeepChem